### DISCUSSION TITLE
Linux (Ubuntu) Coreminer Setup Instructions and Guide for Beginners

### DISCUSSION THREAD

**Discussion by tgh90** [catchthatrabbit/coreminer#83] [Operating System]
### Linux (Ubuntu) Coreminer Setup Instructions and Guide for Beginners

1. Open the Terminal

2. Create a new folder: `mkdir coreminer`

3. Change directory: `cd coreminer`

4. Download the Coreminer software. The current version used in this guide is v0.19.80.
    For a x86_64 CPU it's:
    `wget -P ./ "https://github.com/catchthatrabbit/coreminer/releases/download/v0.19.80/coreminer-linux-x86_64.tar.gz"`
    For an ARM CPU it's:
    `wget -P ./ "https://github.com/catchthatrabbit/coreminer/releases/download/v0.19.80/coreminer-linux-arm64.tar.gz"`

5. Extract coreminer-linux-x86_64.tar.gz: `tar -xf coreminer-linux-x86_64.tar.gz`
    Or for coreminer-linux-arm64.tar.gz: `tar -xf coreminer-linux-arm64.tar.gz`

6. Change directory: `cd coreapp`

7. Install bc: `sudo apt-get install bc`

8. Start the Coreminer: `bash mine.sh`

For the Core Coin (XCB) Wallet go to: [Wallet Generator](https://github.com/core-coin/wallet-generator)

To go outside of a folder you have to use: `cd ..`

You can stop the miner with 2x **Ctrl+C**. Every time you restart the Terminal you just have to change directory `cd coreminer/coreapp/` and start the Coreminer `bash mine.sh`. Also, you can combine them in one command `cd coreminer/coreapp/ && bash mine.sh`.

If you want to change the Coreminer settings you have to make changes in the pool.cfg file in the Coreminer folder. You can open it with e.g. `nano pool.cfg`. To save it hit **Ctrl+X**, **Y**, and then **Enter**. Alternatively, you can also remove `rm pool.cfg` the pool.cfg file and start the Coreminer again.

If there is a new version of the Coreminer ([Releases](https://github.com/catchthatrabbit/coreminer/releases)) navigate outside of the **coreminer** folder and remove it with `rm -r coreminer`. Then repeat this guide but change the version number on step 4 to the newer version or copy the link and paste it between the quotation marks to replace the download link. Use the newer versions at your own risk because you might need to fix some things to make them work.

You don't have to type in every command twice. Just use the arrow keys up and down on your keyboard to navigate through the history of your previously used commands.

To make a software update and reboot use this command:
`sudo killall snap-store && sudo snap refresh & sudo apt update && sudo apt upgrade -y && sudo reboot`
If you run into any trouble with this command just manually reboot and run it again.

**Comment by mrmike15** [comment]
In linux ubuntu how do you add a auto start to it just in case of interuption..

**Reply by tgh90** [reply]
Auto start was added in [v0.19.81](https://github.com/catchthatrabbit/coreminer/releases/tag/v0.19.81).

**Comment by mrmike15** [comment]
Thanks but this only shows 19:80  https://github.com/catchthatrabbit/coreminer/discussions/83
    On Saturday, July 22, 2023 at 12:15:02 PM CDT, tgh90 ***@***.***> wrote:  
 
 


Auto start was added in v0.19.81.
